examples/ds/directory.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: ds.aws.upbound.io/v1beta1
+kind: Directory
+metadata:
+  annotations:
+    upjet.upbound.io/manual-intervention: "This resource needs a valid subnetIds values."
+    meta.upbound.io/example-id: ds/v1beta1/directory
+  labels:
+    testing.upbound.io/example-name: bar
+  name: bar
+spec:
+  forProvider:
+    name: corp.notexample.com
+    passwordSecretRef:
+      key: example-key
+      name: example-secret
+      namespace: upbound-system
+    region: us-east-1
+    size: Small
+    tags:
+      Project: foo
+    vpcSettings:
+      - subnetIds:
+          - "subnet-0dd55c657e320dc88"
+          - "subnet-01d7650bd91693a9f"
```

Comment:
It seems that the auto-reference generator could not resolve these references. So, we may consider adding them manually to config.go. By this way, we can use the Subnet resources as reference here (for example selector reference) and we can use Uptest for this resource

Reply:
@sergenyalcin I had tried to use selectors but, unfortunately, I could't manage it.
Here I asked for advice: https://upboundio.slack.com/archives/C043UMVE3MF/p1670423205754799

Reply:
For using selectors (or reference fields) firstly we need to add the reference manually. In ideal case, reference generator can resolve the references and add them to the API schema. But some corner cases it cannot resolve them. For these type of situations we may add these references with the following statement:

```go
p.AddResourceConfigurator("aws_directory_service_directory", func(r *config.Resource) {
		r.References["subnet_ids"] = config.Reference{
			Type: "github.com/upbound/provider-aws/apis/ec2/v1beta1.Subnet",
		}
	})
```

After adding this to the config.go and generating the schema, you can start to use the selector references.

Reply:
I have created config.go, added dependencies:
`"github.com/upbound/provider-aws/config/ds"` && `ds.Configure`
to `provider.go`. But when I tried to apply the resource I got an error:
```error: error validating "examples/ds/directory.yaml": error validating data: [ValidationError(Directory.spec.forProvider.vpcSettings[0].subnetIds[0]): invalid type for io.upbound.aws.ds.v1beta1.Directory.spec.forProvider.vpcSettings.subnetIds: got "map", expected "string", ValidationError(Directory.spec.forProvider.vpcSettings[0].subnetIds[1]): invalid type for io.upbound.aws.ds.v1beta1.Directory.spec.forProvider.vpcSettings.subnetIds: got "map", expected "string"]; if you choose to ignore these errors, turn validation off with --validate=false```
I tried several variations such as: `subnetIdRef` and `subnetIdSelector` but it couldn't help.
Also I noticed that in `zz_directory_types.go` file `VPCSettingsParameters` section didn't change.
There are attached files
[config.txt](https://github.com/upbound/provider-aws/files/10229045/config.txt)
[directory.txt](https://github.com/upbound/provider-aws/files/10229046/directory.txt)
[zz_directory_types.txt](https://github.com/upbound/provider-aws/files/10229047/zz_directory_types.txt)
maybe it will be helpful.

Reply:
@mykolalosev it's my bad. This field is under the vpc_settings. So we need to add the reference with the following way. When you add and generate you will see some changes in API:

```go
p.AddResourceConfigurator("aws_directory_service_directory", func(r *config.Resource) {
		r.References["vpc_settings.subnet_ids"] = config.Reference{
			Type: "github.com/upbound/provider-aws/apis/ec2/v1beta1.Subnet",
		}
	})
```

Reply:
Yes, it's working. Thank you!